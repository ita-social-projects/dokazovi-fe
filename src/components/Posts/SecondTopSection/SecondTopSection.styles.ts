import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(
  () => ({
    root: {
      marginTop: '50px',
      marginBottom: '50px',
    },
    authorName: {
      fontSize: '24px',
      lineHeight: '1.17',
    },
    authorBio: {
      fontSize: '15px',
      lineHeight: '1.47',
    },
  }),
  { name: 'SecondPostView' },
);
