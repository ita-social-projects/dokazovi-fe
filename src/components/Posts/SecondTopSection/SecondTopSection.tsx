import { Box, Typography } from '@material-ui/core';
import React from 'react';
import { Link } from 'react-router-dom';
import { useStyles } from './SecondTopSection.styles';

interface ISecondTopSection {
  author: {
    firstName: string;
    id: number;
    lastName: string;
    bio?: string;
  };
}

export default function SecondTopSection({
  author,
}: ISecondTopSection): JSX.Element {
  const classes = useStyles();
  const { id, firstName, lastName, bio } = author;
  const authorFullName = `${firstName} ${lastName}`;
  return (
    <Box className={classes.root}>
      <Link to={`/experts/${id}`}>
        <Typography variant="h4" className={classes.authorName}>
          {authorFullName}
        </Typography>
      </Link>
      <Typography color="textSecondary" className={classes.authorBio}>
        {bio}
      </Typography>
    </Box>
  );
}
