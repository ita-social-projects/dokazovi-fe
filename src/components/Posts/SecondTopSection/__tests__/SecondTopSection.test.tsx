import React from 'react';
import { render, screen, RenderResult } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import SecondTopSection, { ISecondTopSection } from '../SecondTopSection';

const secondTopSectionMocks: ISecondTopSection = {
  author: {
    firstName: 'Bill',
    id: 56,
    lastName: 'Yamamoto',
    bio:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  },
  origin: 3,
};

describe('SecondTopSection renders with main block', () => {
  let component: RenderResult;

  beforeEach(() => {
    component = render(
      <BrowserRouter>
        <SecondTopSection
          author={secondTopSectionMocks.author}
          origin={secondTopSectionMocks.origin}
        />
      </BrowserRouter>,
    );
  });

  it('should SecondTopSection component render correctly', () => {
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
  });

  it('should main block be in SecondTopSection component', () => {
    expect(screen.getByTestId('block')).toBeInTheDocument();
  });
});

describe('SecondTopSection renders with origin not equal to 3', () => {
  it('should origin not equal to 3', () => {
    delete secondTopSectionMocks.origin;

    const { asFragment } = render(
      <BrowserRouter>
        <SecondTopSection
          author={secondTopSectionMocks.author}
          origin={secondTopSectionMocks.origin}
        />
      </BrowserRouter>,
    );

    expect(screen.getByTestId('link')).toBeInTheDocument();
    expect(screen.getByTestId('typography')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
