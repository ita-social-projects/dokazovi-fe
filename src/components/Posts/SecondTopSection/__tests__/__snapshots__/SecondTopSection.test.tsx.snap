// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondTopSection renders with main block should SecondTopSection component render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4 SecondPostView-root-1"
    data-testid="block"
  >
    <a
      href="/materials?origins=3"
    >
      <h4
        class="MuiTypography-root SecondPostView-authorName-2 MuiTypography-h4"
      >
        Bill Yamamoto
      </h4>
      <p
        class="MuiTypography-root SecondPostView-authorBio-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </a>
  </div>
</DocumentFragment>
`;

exports[`SecondTopSection renders with origin not equal to 3 should origin not equal to 3 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-12 SecondPostView-root-9"
    data-testid="block"
  >
    <a
      data-testid="link"
      href="/experts/56"
    >
      <h4
        class="MuiTypography-root SecondPostView-authorName-10 MuiTypography-h4"
      >
        Bill Yamamoto
      </h4>
    </a>
    <p
      class="MuiTypography-root SecondPostView-authorBio-11 MuiTypography-body1 MuiTypography-colorTextSecondary"
      data-testid="typography"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </p>
  </div>
</DocumentFragment>
`;
